index.html

```diff
@@ -808,6 +811,90 @@ <h3>
           </p>
         </section>
       </section>
+      <section>
+        <h3>
+          `id` member
+        </h3>
+        <p>
+          The [=manifest's=] <code><dfn data-export="" data-dfn-for=
+          "manifest" id="identity">id</dfn></code> member is a <a>string</a> that
+          represents the <dfn>identity</dfn> for the application. 
+          The |identity| takes the form of a URL, which is same origin as the
+          start URL.
+        </p>
+        <p>The |identity| is used by user agents to uniquely identify the
+        application universally. When the user agent sees a manifest with an
+        |identity| that does not correspond to an already-installed application,
+        it SHOULD treat that manifest as a description of a distinct
+        application, even if it is served from the same URL as that of another
+        application. When the user agent sees a manifest with an |identity|
+        matching that of an already-installed application, it SHOULD be used as
+        a signal that this manifest is a replacement for the already-installed
+        application's manifest, and not a distinct application, even if it is
+        served from a different URL than the one seen previously.
+        </p>
+        <p>
+          The |identity| MAY be used by a service that collects lists of web
+          applications to uniquely identify applications.
+        </p>
+        <p>
+          To <dfn>process the `id` member</dfn>, given [=object=]
+          |json:JSON|, [=ordered map=] |manifest:ordered map|:
+        </p>
+        <ol class="algorithm">
+          <li>Set |manifest|["id"] to |manifest|["start_url"].
+          </li>
+          <li>If the type of |json|["id"] is not [=string=], return.
+          </li>
+          <li>Let |id:URL| be the result of [=URL Parser|parsing=] |json|["id"]
+          with |manifest|["start_ur"]'s origin as the base URL.
+          </li>
+          <li>If |id| is not same origin as |manifest|["start_url"], return.
+          </li>
+          <li>If |id| is failure, return.
```

Comment:
hmm... the algorithm currently doesn't quite do as intended... here is the JS equivalent of what's currently specified:

```JS
/**
 * @param {Object} json
 * @param {Map<string, any>} manifest
 */
function processIdMember(json, manifest){
  // Set |manifest|["id"] to |manifest|["start_url"].
  manifest.set("id", manifest.get("start_url"));

  // If the type of |json|["id"] is not [=string=], return.
  if (typeof json.id !== "string") return;

  // Let |id:URL| be the result of [=URL Parser|parsing=] |json|["id"]
  // with |manifest|["start_ur"]'s origin as the base URL.
  const id = new URL(json.id, manifest.get("start_url"));

  // If |id| is not same origin as |manifest|["start_url"], return.
  if(id.origin !== manifest.get("start_url").origin)) return;

  // If |id| is failure, return.
}
```

I think what we want is:

```JS
/**
 * @param {Object} json
 * @param {Map<string, any>} manifest
 */
 function processIdMember(json, manifest) {
  const startURL = manifest.get("start_url");
  const fallbackId = new URL(startURL);
  let id;
  if (typeof json.id === "string") {
    try {
      id = new URL(json.id, startURL);
    } catch {
      // It failed, so use the fallbackId
      id = fallbackId;
    }
  }

  if (id.origin !== startURL.origin) {
    // user agent would warn here...
    id = fallbackId;
  }
  // finally, we have the id...
  manifest.set("id", id);
}
```

Reply:
Maybe more like:
```js
/**
 * @param {Object} json
 * @param {Map<string, any>} manifest
 */
 function processIdMember(json, manifest) {
  const startURL = manifest.get("start_url");
  const fallbackId = new URL(startURL);
  if (typeof json.id !== "string") {
    manifest.set("id", fallbackId);
    return;
  }

  // Maye we need to parse the json id as a url first?
  if (json.id.origin !== startURL.origin) {
    manifest.set("id", fallbackId);
    return;
  }

  let id;
  try {
      id = new URL(json.id, startURL.origin);
   } catch {
      id = fallbackId;
   }

  // finally, we have the id...
  manifest.set("id", id);
}
```

Reply:
I updated the spec to add a final statement of "Otherwise, set manifest["id"] to |id|" And reordered the same origin check and URL processing result check.
It should match 
```javascript
/**
 * @param {Object} json
 * @param {Map<string, any>} manifest
 */
function processIdMember(json, manifest){
  // Set |manifest|["id"] to |manifest|["start_url"].
  manifest.set("id", manifest.get("start_url"));

  // If the type of |json|["id"] is not [=string=], return.
  if (typeof json.id !== "string") return;

  // Let |id:URL| be the result of [=URL Parser|parsing=] |json|["id"]
  // with |manifest|["start_ur"]'s origin as the base URL.
  let id;
  try {
    id = new URL(json.id, manifest.get("start_url").origin);
  } catch {
   // if |id| is failure, return.
    return;
  }

  // If |id| is not same origin as |manifest|["start_url"], return.
  if(id.origin !== manifest.get("start_url").origin)) return;
  
  //Otherwise, set |manifest|["id"] to |id|
  manifest.set("id", id);
}
```